plugins/hikvision/src/main.ts

```diff
@@ -478,12 +488,76 @@ export class HikvisionCamera extends RtspSmartCamera implements Camera, Intercom
         if (this.hasSmartDetection)
             interfaces.push(ScryptedInterface.ObjectDetector);
 
-        if (this.hasAlarm() || this.hasSupplementLight()) 
+        if (this.hasAlarm() || this.hasSupplementLight())
```

Comment:
ugh I'm just now noticing these are async calls that will always return true because they're awaited. the previous pull request was also a mistake.

refer to the implementation of hasSmartDetection that caches the capability state.

The issue with this change is that it will flap interfaces during plugin reload since hasPtz starts out as false.